endpoints/openrtb2/auction_test.go

```diff
@@ -865,6 +865,52 @@ func validRequest(t *testing.T, filename string) string {
 	return string(requestData)
 }
 
+func TestCurrencyTrunc(t *testing.T) {
+	deps := &endpointDeps{
+		&nobidExchange{},
+		newParamsValidator(t),
+		&mockStoredReqFetcher{},
+		empty_fetcher.EmptyFetcher{},
+		empty_fetcher.EmptyFetcher{},
+		&config.Configuration{},
+		pbsmetrics.NewMetrics(metrics.NewRegistry(), openrtb_ext.BidderList(), config.DisabledMetrics{}),
+		analyticsConf.NewPBSAnalytics(&config.Analytics{}),
+		map[string]string{},
+		false,
+		[]byte{},
+		openrtb_ext.BidderMap,
+	}
+
+	ui := uint64(1)
+	req := openrtb.BidRequest{
+		ID: "someID",
+		Imp: []openrtb.Imp{
+			{
+				ID: "imp-ID",
+				Banner: &openrtb.Banner{
+					W: &ui,
+					H: &ui,
+				},
+				Ext: json.RawMessage("{\"appnexus\": {\"placementId\": 5667}}"),
+			},
+		},
+		Site: &openrtb.Site{
+			ID: "myID",
+		},
+		Cur: []string{"USD", "EUR"},
+	}
+
+	errL := deps.validateRequest(&req)
+	if len(errL) > 0 {
+		for _, err := range errL {
+			if err.Error() != "A prebid request can only process one currency. Taking the first currency in the list, USD, as the active currency" {
+				t.Errorf(err.Error())
+			}
```

Comment:
Would be good to assert the error code and the error type as well for the multiple currency warning

Reply:
That may be a bit overkill. A lot would have to go wrong to get an error with that message, but somehow not have the correct code/type.

Reply:
Well, I was suggesting for those assertions more to safeguard us from situations where somebody touching that code and potentially changing the error type for it, from `warning` to something else, maybe a fatal error type, then the tests can help catch that.

Reply:
If someone just randomly changes the error type, that is a pretty blatant and silly change. If it something more subtle like the error code changing, then the test will probably target the new code, unless something really subtle is going on. In any case, I had already beefed up the test since it wasn't much work.